python_ta/debug/snapshot.py

```diff
@@ -29,11 +34,17 @@ def get_filtered_global_variables(frame: FrameType) -> dict:
     return {"__main__": true_global_vars}
 
 
-def snapshot():
+def snapshot(
+    save: Optional[bool] = False,
```

Comment:
You're misunderstanding the `Optional` type annotation; it doesn't mean "optional parameter", it means "could be `None`". So you should use `Optional` for memory_viz_args, but not the other two.